Odata-docs/webapi-8/tutorials/custom-skiptokenhandler.md

```diff
@@ -0,0 +1,549 @@
+
+## Paging using the default skip token handler
+
+By default, ASP.NET.Core OData uses a default implementation of `SkipTokenHandler` called `DefaultSkipTokenHandler`. This generates skip tokens by encoding keys and values of the last resource in the response.
+
+Let's execute the following request to see an example:
+
+```http
+GET /odata/Customers
+```
+
+**Response**:
+
+```json
+{
+    "@odata.context": "http://localhost:5000/odata/$metadata#Customers",
+    "value": [
+        {
+            "Id": 1,
+            "Name": "Customer 1"
+        },
+        {
+            "Id": 2,
+            "Name": "Customer 2"
+        }
+    ],
+    "@odata.nextLink": "http://localhost:5000/odata/Customers?$skiptoken=Id-2"
+}
+```
+
+Notice the skip token `Id-2`, this was generated by the `DefaultSkipTokenHandler` based on the last customer in the resource. It encodes the field `Id` and value `2`. This assumes that the `Id` field has unique value for each customer, it also assumes the collection has a natural order.
+
+If you order the results by a different field, this will also be encoded in the skip token.
+
+For example, let's order the results by name, in descending order:
+
+```http
+GET /odata/Customers?$orderby=Name desc
+```
+
+> [!NOTE]
+> Ensure you enable the `$orderby` query option in your service configuration using `options.OrderBy()` or `options.EnableQueryFeatures(null)`
+
+**Response:**
+
+```json
+{
+    "@odata.context": "http://localhost:5000/odata/$metadata#Customers",
+    "value": [
+        {
+            "Id": 5,
+            "Name": "Customer 5"
+        },
+        {
+            "Id": 4,
+            "Name": "Customer 4"
+        }
+    ],
+    "@odata.nextLink": "http://localhost:5000/odata/Customers?$orderby=Name%20desc$skiptoken=Name-%27Customer%205%27,Id-4"
+}
+```
+
+In this case, the skip token encodes both the name and Id of the last customer in the response. It will use the name to determine which customer should appear first in the next page, and if there are customers with the same name, it will use the ID as a tie-breaker.
+
+> [!IMPORTANT]
+> You should not rely on the `DefaultSkipTokenHandler` generating the skip token in a specific format. We make no guarantee that this format may not change between library versions.
+
+If you wish to change how skip tokens are generated or how they're processed, you can replace the `DefaultSkipTokenHandler` with your own custom implementation of `SkipTokenHandler`.
+
+## Creating a custom skip token handler
+
+The `SkipTokenHandler` is response for generating the *next link* in an OData response when using server-driven paging. It's also responsible for applying the `$skiptoken` to the results. `SkipTokenHandler` is an abstract class that defines the following methods to be implemented by child classes:
```

Comment:
```suggestion
The `SkipTokenHandler` is responsible for generating the *next link* in an OData response when using server-driven paging. It's also responsible for applying the `$skiptoken` to the results. `SkipTokenHandler` is an abstract class that defines the following methods to be implemented by child classes:
```

Reply:
or The `SkipTokenHandler` generates the *next link* in an OData response when using server-driven paging